rbac/management/management/commands/sync_schemas.py

```diff
@@ -0,0 +1,148 @@
+#
+"""Schema sync command."""
+
+from django.core.management.base import BaseCommand
+from django.db.utils import IntegrityError
+from management.models import Group, Policy, Principal, Role
+from tenant_schemas.utils import tenant_context
+
+from api.models import Tenant
+
+
+class Command(BaseCommand):
+    """Command for use with django's manage.py script."""
+
+    help = "Syncs custom principals, roles, groups, and policies to the public schema"
+
+    def clear_pk(self, entry):
+        """Clear the ID and PK values for provided postgres entry."""
+        entry.id = None
+        entry.pk = None
+
+    def copy_custom_principals_to_public(self, tenant):
+        """Copy custom principals from provided tenant to the public schema."""
+        principals = Principal.objects.all()
+        public_schema = Tenant.objects.get(schema_name="public")
+        for principal in principals:
+            self.stdout.write(f"Copying principal {principal.username} to public schema for tenant {tenant}.")
+            if not principal.tenant:
+                principal.tenant = tenant
+                principal.save()
+            with tenant_context(public_schema):
+                self.clear_pk(principal)
+                try:
+                    principal.save()
+                except IntegrityError as err:
+                    self.stderr.write(f"Couldn't copy principal: {principal.username}. Skipping due to:\n{err}")
+                    continue
+
+    def copy_custom_roles_to_public(self, tenant):
+        """Copy custom roles from provided tenant to the public schema."""
+        roles = Role.objects.filter(system=False)
+        public_schema = Tenant.objects.get(schema_name="public")
+        for role in roles:
+            self.stdout.write(f"Copying role {role.name} to public schema for tenant {tenant}.")
+            if not role.tenant:
+                role.tenant = tenant
+                role.save()
+            access_list = list(role.access.all())
+            with tenant_context(public_schema):
+                self.clear_pk(role)
+                try:
+                    role.save()
+                except IntegrityError as err:
+                    self.stderr.write(f"Couldn't copy role: {role.name}. Skipping due to:\n{err}")
+                    continue
+                for access in access_list:
+                    self.clear_pk(access)
+                    if not access.tenant:
+                        access.tenant = tenant
+                    access.role = role
+                    try:
+                        access.save()
+                    except IntegrityError as err:
+                        self.stderr.write(f"Couldn't copy access entry: {access}. Skipping due to:\n{err}")
+                        continue
+                    for resource_def in access.resourceDefinitions.all():
+                        self.clear_pk(resource_def)
+                        resource_def.access = access
```

Comment:
Should we set the tenant_id on these as well, similar to the other models?

Reply:
This actually gets set in line 74 above.